browser/src/control/Control.Menubar.js

```diff
@@ -2002,6 +2002,10 @@ L.Control.Menubar = L.Control.extend({
 	_createMenu: function(menu) {
 		var itemList = [];
 		var docType = this._map.getDocType();
+		var isReadOnly = this._map.isPermissionReadOnly();
+		if (docType !== 'drawing' && isReadOnly) {
```

Comment:
We should check probably if it is .pdf, so it will not appear in readonly .odg, but probably it is not possible here directly (on client side).

We have 'app.file.fileBasedView' variable which indicates that we don't watch separate slides but have continuous scrolling - which is used in PDFs. Maybe that one we should use? I think @gokaysatir knows if that is enough condition.

Reply:
@eszkadev makes sense. What do you think @gokaysatir ? And what about `BaseFileName` https://github.com/CollaboraOnline/online/blob/master/browser/src/map/handler/Map.WOPI.js#L102 ?

Reply:
I have tried with `app.file.fileBasedView` but it ends up being true on pdf, impress and drawing when readonly-mode:
```
false [Control.Menubar.js:2006:10](localhost:9980/browser/src/control/Control.Menubar.js)
true [Control.Menubar.js:2006:10](localhost:9980/browser/src/control/Control.Menubar.js)
true [Control.Menubar.js:2006:10](localhost:9980/browser/src/control/Control.Menubar.js)
```

Reply:
I know we avoid depend on file extension so I wonder if it's not better to check if all children are disabled and if it is add it to the hidden array:
The problem is that the direct children do not have disabled status: 
```javascript
Array(8) [ li#menu-saveas, li#menu-savecomments, li#menu-shareas, li#menu-print, li#menu-rev-history, li#menu-downloadas, li, li#menu-closedocument.readonly ]
​
0: <li id="menu-saveas" class="" role="menuitem">
​
1: <li id="menu-savecomments" class="" role="menuitem">
​
2: <li id="menu-shareas" class="" role="menuitem">
​
3: <li id="menu-print" class="" role="menuitem">
​
4: <li id="menu-rev-history" class="" role="menuitem">
​
5: <li id="menu-downloadas" class="" role="menuitem">
​
6: <li class="" role="menuitem">
​
7: <li id="menu-closedocument" class="readonly" role="menuitem">
​
length: 8
​
<prototype>: Array []
Control.Menubar.js:2076:10
```

just one level under we have that info in the `<a>`

Reply:
@gokaysatir 
> filebasedview shouldn't be set to true on read only mode. It seems to be a bug to me.

Reply:
I think so

Reply:
I have an idea to detect file type according to mimetype. We are detecting docType here for Impress
ImpressTileLayer.js: _onStatusMsg: function (textMsg)
  var command = app.socket.parseServerCmd(textMsg)
  this._docType = command.type;

If I'm correct, server sends that textMsg that icludes "type" info. If we send the mime type too from server side to online, we can define a getDocMimeType() Is there any obstacle for us to do this?

Reply:
yes, we can use that if it is already sent.
Only we have to check if ALWAYS `createMenu` occurs after we receive status message. In other words: if we create menubar after we received status message (eg. in `initializeSpecializedUI` in `UIManager`) or before/concurrently (in `initializeBasicUI`). So we will be sure there is no race here.

Reply:
As I learn the max info about the document as following:
    void dumpState(std::ostream& oss)
    {
        oss << "Kit Document:\n"
            << std::boolalpha
            << "\n\tpid: " << getpid()
            << "\n\tstop: " << _stop
            << "\n\tjailId: " << _jailId
            << "\n\tdocKey: " << _docKey
            << "\n\tdocId: " << _docId
            << "\n\turl: " << _url
            << "\n\tobfuscatedFileId: " << _obfuscatedFileId
            << "\n\tjailedUrl: " << _jailedUrl
            << "\n\trenderOpts: " << _renderOpts
            << "\n\thaveDocPassword: " << _haveDocPassword // not the pwd itself
            << "\n\tisDocPasswordProtected: " << _isDocPasswordProtected
            << "\n\tdocPasswordType: " << (int)_docPasswordType
            << "\n\teditorId: " << _editorId
            << "\n\teditorChangeWarning: " << _editorChangeWarning
            << "\n\tmobileAppDocId: " << _mobileAppDocId
            << "\n\tinputProcessingEnabled: " << _inputProcessingEnabled
            << "\n";

Extra we know WOPISrc the local path of the document. For xml based filetypes we can reach the "mimetype" if we unzip the files and read the xml files. But for PDF is not xml based and couldn't find if it includes any mimetype info.

It seems to me we have to rely file extension.